creator-node/src/fileManager.js

```diff
@@ -145,31 +166,39 @@ async function saveFileForMultihashToFS (req, multihash, expectedStoragePath, ga
     req.logger.debug(`checking if ${multihash} already available on local ipfs node`)
     try {
       // ipfsCat returns a Buffer
+      decisionTree.push({ stage: 'About to retrieve file from local ipfs node with cat', vals: multihash, time: Date.now() })
       let fileBuffer = await Utils.ipfsCat(multihash, req, 1000)
       fileFound = true
-      req.logger.debug(`Retrieved file for ${multihash} from local ipfs node`)
+      req.logger.debug(`Retrieved file for ${multihash} from  with cat`)
+      decisionTree.push({ stage: 'Retrieved file from local ipfs node with cat', vals: multihash, time: Date.now() })
       // Write file to disk.
       await writeFile(expectedStoragePath, fileBuffer)
       req.logger.info(`wrote file to ${expectedStoragePath}, obtained via ipfs cat`)
+      decisionTree.push({ stage: 'Wrote file to disk', vals: expectedStoragePath, time: Date.now() })
     } catch (e) {
       req.logger.warn(`Multihash ${multihash} is not available on local ipfs node ${e.message}`)
+      decisionTree.push({ stage: 'File not available on local ipfs node with cat', vals: multihash, time: Date.now() })
     }
 
     // If file not already available on local ipfs node, fetch from IPFS.
     if (!fileFound) {
       req.logger.debug(`Attempting to get ${multihash} from IPFS`)
       try {
+        decisionTree.push({ stage: 'About to retrieve file from local ipfs node with get', vals: multihash, time: Date.now() })
         // ipfsGet returns a BufferListStream object which is not a buffer
         // not compatible into writeFile directly, but it can be streamed to a file
         let fileBL = await Utils.ipfsGet(multihash, req, 5000)
         req.logger.debug(`retrieved file for multihash ${multihash} from local ipfs node`)
+        decisionTree.push({ stage: 'Retrieved file from local ipfs node with get', vals: multihash, time: Date.now() })
 
         // Write file to disk.
         await Utils.writeStreamToFileSystem(fileBL, expectedStoragePath)
         fileFound = true
         req.logger.info(`wrote file to ${expectedStoragePath}, obtained via ipfs get`)
+        decisionTree.push({ stage: 'Wrote file to disk', vals: expectedStoragePath, time: Date.now() })
       } catch (e) {
         req.logger.warn(`Failed to retrieve file for multihash ${multihash} from IPFS ${e.message}`)
+        decisionTree.push({ stage: 'File not available on local ipfs node with ipfs get', vals: multihash, time: Date.now() })
```

Comment:
there are a few places where these log statements are now redundant with the eventual decision tree log. do you want to remove? can see argument for keeping as well. your call, just flagging

Reply:
i felt we should just add this one statement net new and we can peel back everything else after the fact?

Reply:
sg